### DISCUSSION TITLE
Meshing a truncated cyliner by extruding a surface mesh

### DISCUSSION THREAD

**Discussion by ews-ffarella** [mdolab/pyhyp#82] [Q&A] [answered]
Hello,
I would like to generate a mesh based by extruding a surface mesh (topography). The final volume should be a vertical cylinder (truncated by the input surface mesh), but it could also be a cube. This is for a simulation of the atmospheric boundary layer. 

see example 

![image](https://github.com/mdolab/pyhyp/assets/13105859/b4fc8105-9f01-4cd8-8cb9-e772e130a17c)

![image](https://github.com/mdolab/pyhyp/assets/13105859/64f23b0e-72ed-43e5-afd3-f48608c7800b)

Could you give me pointers about how to do it? I guess I will use the python interface. 

Also, would it be possible to perform the extrusion in 2 steps. Once with a constant stretch factor (close to 1), and then with a larger one?

Thank you very much for providing the community with such a great tool ;)

**Comment by sseraj** [comment]
Unfortunately, pyHyp cannot handle this use case because the farfield boundary shape cannot be prescribed when using hyperbolic marching.

**Reply by ews-ffarella** [reply]
Thanks for the reply. What would be the shape of the produced mesh?

**Reply by sseraj** [reply]
The farfield will look like a smoothed version of the input surface. For our most common use cases, the farfield looks spherical because we extrude far away from the surface.

**Reply by ews-ffarella** [reply]
Thank you! I might still give it a try and constrain the mesh manually after. The normals at the boundaries are pointing strictly upwards, so it might help. 

Would this example be the closest to what I need (3d extrusion)?

https://github.com/mdolab/pyhyp/blob/main/examples/sphere/runSphere.py

**Reply by sseraj** [reply]
I think this example might be closer to what you are looking for https://github.com/mdolab/pyhyp/blob/main/examples/plate/runPlate.py. You should be able to set the `BC` option to keep the sides of the mesh cylindrical.

**Reply by ews-ffarella** [reply]
Thanks, I have played around with the plate example. 

<s>I have tried with an STL converted to CGNS using paraview, but  it get an  `cgio_open_file:invalid file type` error.
Does the input geometry need to be all quads?</s>

OK, I think you can only get structured meshes and input, right?

![image](https://github.com/mdolab/pyhyp/assets/13105859/c0646b4b-a1d1-429f-b071-50702809c756)

**Comment by ews-ffarella** [comment, marked as answer]
Amazing!

![image](https://github.com/mdolab/pyhyp/assets/13105859/d6a6a1cd-a781-4738-b02d-3c9bb35a407c)


With full control of the 2D grid:

![image](https://github.com/mdolab/pyhyp/assets/13105859/b4305aa1-406a-4887-92aa-c9ec00eaf80a)

![Uploading image.png…]()


You have no idea how long I had been searching for this. Thank you soooooo much 😄
